### PR TITLE
Make default test pattern narrower

### PR DESCRIPTION
**Pull request by nohwnd**
**Task name**: VSTestTask2

**Description**: Change the default pattern: 

VSTest2 task specifies a very wide dll matching pattern which includes many unexpected dlls. This includes all TestPlatform dlls,
test framework dlls, and often dlls from local nuget package cache. This is slowing TestPlatform down, and highlights issue which was previously (in TestPlatform 17.2) silently ignored.

Instead a narrow pattern that matches dlls from bin folder only should be suggested as the default, as it fits most test projects we see in the wild and matches the common test dll naming pattern such as:

```
Product.Test.dll
Product.Tests.dll
ProductUnitTests.dll
Product.Unit.Tests.dll
```

It also does not match `*TestPlatform*.dll` or `MSTest.TestFramework.dll`, so no additional exclude patterns are necessary.

This leads to more correct test runs, and faster test runs as less dlls are included.

Related https://github.com/microsoft/vstest/issues/3939

**Documentation changes required:** Y

Fixed it in the readme, could not find it anywhere else in this repo.

**Added unit tests:** N

I could not find any, will fix them if the build does not pass.

**Attached related issue:** Y

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

Here is a build using the same pattern: https://nohwnd.visualstudio.com/nunit/_build/results?buildId=2144&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=7bfb7e0f-549d-5929-c603-6e8046613b9e

### CODE REVIEW

**Comment by nohwnd** [comment]
A poll here to see if `common test dll naming pattern` match what I regularly see:

https://twitter.com/nohwnd/status/1560560403104821248

File: Tasks/VsTestV2/task.loc.json [hunk 1]
```diff
@@ -69,7 +69,7 @@
       "name": "testAssemblyVer2",
       "type": "multiLine",
       "label": "ms-resource:loc.input.label.testAssemblyVer2",
-      "defaultValue": "**\\*test*.dll\n!**\\*TestAdapter.dll\n!**\\obj\\**",
+      "defaultValue": "**\\bin\\**\\test.dll\n**\\bin\\**\\tests.dll\n",
```
Comment (triptijain2112) [inline Tasks/VsTestV2/task.loc.json:72]: It should be *test.dll and *tests.dll, we can copy from task.json file above. That looks good

Comment (nohwnd) [reply Tasks/VsTestV2/task.loc.json:72]: The new filter says
```suggestion
      "defaultValue": "**\\bin\\**\\test.dll\n**\\bin\\**\\tests.dll",
```

Comment (triptijain2112) [reply Tasks/VsTestV2/task.loc.json:72]: image
Is this difference expected?

Comment (nohwnd) [reply Tasks/VsTestV2/task.loc.json]: It is not, you are right, I am missing stars there, fixing.

File: Tasks/VsTestV2/task.loc.json [hunk 2]
```diff
@@ -69,7 +69,7 @@
       "name": "testAssemblyVer2",
       "type": "multiLine",
       "label": "ms-resource:loc.input.label.testAssemblyVer2",
-      "defaultValue": "**\\*test*.dll\n!**\\*TestAdapter.dll\n!**\\obj\\**",
+      "defaultValue": "**\\bin\\**\\test.dll\n**\\bin\\**\\tests.dll",
```
Comment (nohwnd) [inline Tasks/VsTestV2/task.loc.json:72]:
```suggestion
      "defaultValue": "**\\bin\\**\\*test.dll\n**\\bin\\**\\*tests.dll",
```

File: Tasks/VsTestV2/task.json [hunk 3]
```diff
@@ -17,7 +17,7 @@
     "author": "Microsoft Corporation",
     "version": {
         "Major": 2,
-        "Minor": 205,
+        "Minor": 209,
```
Comment (triptijain2112) [inline Tasks/VsTestV2/task.json:20]: current sprint i.e. 224

**Comment by nohwnd** [comment]
Merged main, updated your remarks. Created PR for documentation, and linked this PR.

**Review by triptijain2112 — APPROVED** [review]

**Comment by nohwnd** [comment]
@triptijain2112 the build seems to be broken / stuck, could you please re-run it?

**Pull request merged**